rp2040-hal/src/dma/single_buffer.rs

```diff
@@ -112,4 +112,39 @@ where
 
         (self.ch, self.from, self.to)
     }
+
+    /// Aborts the current transfer, returning the channel and targets
+    pub fn abort(mut self) -> (CH, FROM, TO) {
+        let irq0_was_enabled = self.ch.is_enabled_irq0();
+        let irq1_was_enabled = self.ch.is_enabled_irq1();
+        self.ch.disable_irq0();
+        self.ch.disable_irq1();
+
+        // Safety: We only use the atomic alias of the register.
```

Comment:
Is this really true? As I understand it, `chan_abort.write(|w| unsafe { w.chan_abort().bits(abort_mask) });` translates to a direct (volatile) write to the CHAN_ABORT register.

This _may_ be harmless. The datasheet isn't clear about what happens if you write a 0 to a bit in this register (ie. if it overwrites a previously set abort bit on some other channel). But even if this happens to be harmless, I think the safety comment is still misleading.

I think the quoted line should use `write_bitmask_set`.

Reply:
It's another one of the registers that is labelled SC in the datasheet, but violates that contract:
"The bit will remain high until any in-flight transfers have been flushed through the address and data FIFOs"
https://github.com/raspberrypi/pico-sdk/issues/1466

I'd be inclined to err on the side of caution with this register, if only because of the other caveat in the datasheet:
"After writing, this register must be polled until it returns all-zero. Until this point, it is unsafe to restart the channel"

My naive intuition is that you should only need to wait for the channel you care about, not all-zero, so this is already surprising information.

> I think the quoted line should use write_bitmask_set.

To do the above to-the-letter this code should not be re-entrant, and we should only ever write to it when it is already all zeroes (so plain write/write_with_zero should suffice).
Should we wrap this in some kinda of lock (critical-section)?

Reply:
A critical section is only appropriate if we are sure that the operation strictly bounded to a few processor cycles. Otherwise it could cause unacceptable latency for interrupt handling. I'm not sure how long it could take, worst case, for the flag to get cleared.

Reply:
The C SDK also does a simple write to the register:
https://github.com/raspberrypi/pico-sdk/blob/master/src/rp2_common/hardware_dma/include/hardware/dma.h#L527-L532
I think we would be fine doing the same, only the safety comment should be updated. What about:
```
// Safety: While we are not sure if this write call clears ABORT bits for other channels, we don't really care about these bits as we check for BUSY anyway, due to RP2040-E13. The C SDK works the same way.
```

Reply:
Thank you for pointing this out, I was under the impression that it was due to how the comment was used in other places. I've addressed this now.

Sorry for taking a week to get around to this, I have been sick.

Reply:
Don't worry, even without giving a reason it's totally fine to contribute to open source projects at whatever pace suits you best. I hope you are well again!